### PR TITLE
EchoExpress riview

### PR DESCRIPTION
**Pull request by aikshit123**
made  a blog website using mern stack.

### CODE REVIEW

File: EchoExpress/app.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    }
+    next();
+});
+
+const mongoString = process.env.DATABASE_URL;
+
+mongoose.connect(mongoString);
+const database = mongoose.connection;
+
+app.use('/blogs', routes);
```
Comment (rumit0406) [inline EchoExpress/app.js:24]: routes should be named as blogRouter

File: EchoExpress/app.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+const bodyParser = require("body-parser");
+const app = express();
+const routes = require('./routes/routes');
+const mongoose = require('mongoose');
+require('dotenv').config();
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }));
+
+app.use((req, res, next) => {
```
Comment (rumit0406) [inline EchoExpress/app.js:12]: what is the use of this?

File: EchoExpress/model/Blog.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
+const mongoose = require('mongoose');
+
+const Blog = new mongoose.Schema({
+    blogTitle : {
```
Comment (rumit0406) [inline EchoExpress/model/Blog.js:4]: name it as title
since it's under model Blog 
naming BlogTitle is redundant

File: EchoExpress/model/Blog.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        type : String,
+        required: true
+    },
+    
+    blogContent : {
```
Comment (rumit0406) [inline EchoExpress/model/Blog.js:9]: name it as content

File: EchoExpress/model/Blog.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        type: String, 
+        required: true       
+    },
+    
+    blogComment: {
```
Comment (rumit0406) [inline EchoExpress/model/Blog.js:14]: name it as comments

File: EchoExpress/model/Blog.js [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        type: [mongoose.Types.ObjectId],
+        required: false,
+        ref : 'Comment'
+    }
+})
+
+const blogComment = new mongoose.Schema({
```
Comment (rumit0406) [inline EchoExpress/model/Blog.js:21]: should be BlogComment and not blogComment

**Comment by rumit0406** [comment]
naming convention

Get all blogs: GET /blogs
Get particular blog: GET /blogs/:blogId
Update a blog: PUT /blogs/:blogId
Add a blog: POST /blogs
Delete a blog: DELETE /blogs/:blogId

Get all comments of a blog: GET /blogs/:blogId/comments
Add a comment to a blog: POST /blogs/:blogId/comments
Update a comment of a blog: PUT /blogs/:blogId/comments/:commentId
Delete a comment of a blog: DELETE /blogs/:blogId/comments/:commentId

File: EchoExpress/routes/routes.js [hunk 7]
```diff
@@ -0,0 +1,123 @@
+const express = require('express');
+const router = express.Router();
+module.exports = router;
+const Models = require('../model/Blog');
+
+const Blog = Models.Blog;
+const Comment = Models.blogComment;
+
+router.post('/post', async (req, res) => {
```
Comment (rumit0406) [inline EchoExpress/routes/routes.js:9]: fix naming for all api endpoints

File: EchoExpress/routes/routes.js [hunk 8]
```diff
@@ -0,0 +1,123 @@
...
+        console.log(updatedData);
+        const result = await Blog.findByIdAndUpdate(id, {blogContent : updatedData}, {new : true})
+        res.send(result);
+    }
+    catch(error){
+        res.status(400).json({message:error.message})
+    }
+})
+
+router.delete('/delete/:id', async (req, res) => {
```
Comment (rumit0406) [inline EchoExpress/routes/routes.js:67]: logic is of deleting comment but api endpoint here is deleting blog
fix this

**Review by rumit0406 — CHANGES_REQUESTED** [review]
pls address the comments

File: EchoExpress/app.js (hunk 1 above)
Comment (aikshit123) [reply EchoExpress/app.js:24]: done

File: EchoExpress/app.js (hunk 2 above)
Comment (aikshit123) [reply EchoExpress/app.js:12]: remove krke chlgya krdiya remove

File: EchoExpress/model/Blog.js (hunk 3 above)
Comment (aikshit123) [reply EchoExpress/model/Blog.js:4]: blogtitle - title
blogcontent- content
krdiya blogcomment ko comment nhi krskta comment keyword already bhooot jgah use horha hai toh blogcomment ko or kya naam deskta hu

File: EchoExpress/model/Blog.js (hunk 4 above)
Comment (aikshit123) [reply EchoExpress/model/Blog.js:9]: done

File: EchoExpress/model/Blog.js (hunk 5 above)
Comment (aikshit123) [reply EchoExpress/model/Blog.js:14]: done

File: EchoExpress/model/Blog.js (hunk 6 above)
Comment (aikshit123) [reply EchoExpress/model/Blog.js:21]: aisa kya

Comment (aikshit123) [reply EchoExpress/model/Blog.js:21]: isko maine ab comments krdiya

File: EchoExpress/routes/routes.js (hunk 7 above)
Comment (aikshit123) [reply EchoExpress/routes/routes.js:9]: done

File: EchoExpress/routes/routes.js (hunk 8 above)
Comment (aikshit123) [reply EchoExpress/routes/routes.js:67]: voh blog delete krne ka hi hai comment delete krne ka abhi daala nhi

**Pull request merged**